arch/risc-v/src/common/riscv_setjmp.S

```diff
@@ -0,0 +1,83 @@
+############################################################################
+# arch/risc-v/src/common/riscv_setjmp.S
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
```

Comment:
This is part of libc.  It belongs in libs/libc/machine/risc-v so that it can be accessed in user space.

If it is in arch/risc-v, then application space programs will not be able to use it unless in certain protected modes.  That is the purpose of libs/libc/machine:  To permit user space access to CPU-specific library functions.

Reply:
I see that this has already been merged.  That was a bad merge.  The code is flawed because of where it has been positioned in the source tree.

Reply:
Yes, it's better to move all setjmp/longjmp to libc/:
arch/arm/src/armv7-m/gnu/arm_setjmp.S
arch/arm/src/armv8-m/arm_setjmp.S
arch/sim/src/sim/up_setjmp32.S
arch/sim/src/sim/up_setjmp64.S
arch/sim/src/sim/up_setjmp_arm.S

Reply:
> Yes, it's better to move all setjmp/longjmp to libc/

I think currently in PROTECTED and KERNEL modes, it will draw the setjmp/longjmp from arch/ for the kernel portion of the build, but will bring in the newlib implementation in for applications.  That would defeat the purpose of having a custom setjmp/longjmp.